tests/blocks.rs

```diff
@@ -113,22 +112,135 @@ mod test_rand {
+                    conflict_reason: ConflictReason::None,
+                    white_flag_index: i as u32,
+                },
+            )
+        };
 
-        db.drop().await.unwrap();
+        let blocks = std::iter::repeat_with(|| (BlockId::rand(), Block::rand_no_payload_with_parents(parents.clone())))
+            .take(5)
+            .inspect(|(block_id, _)| {
+                children.insert(*block_id);
+            })
+            .enumerate()
+            .map(f)
+            .chain(
+                std::iter::repeat_with(|| (BlockId::rand(), Block::rand_no_payload()))
+                    .take(5)
+                    .enumerate()
+                    .map(f),
+            )
+            .collect::<Vec<_>>();
+
+        collection.insert_blocks_with_metadata(blocks.clone()).await.unwrap();
+        assert_eq!(collection.count().await.unwrap(), 10);
+
+        let mut s = collection.get_block_children(&parents[0], 100, 0).await.unwrap();
+
+        while let Some(child_id) = s.try_next().await.unwrap() {
+            assert!(children.remove(&child_id))
+        }
+        assert!(children.is_empty());
+
+        teardown(db).await;
+    }
+
+    #[tokio::test]
+    async fn test_spending_transaction() {
+        let (db, collection) = setup::<BlockCollection>("test-spending-transaction").await;
+
+        let ctx = bee_block_stardust::protocol::protocol_parameters();
+
+        let block_id = BlockId::rand();
+        let (block, input) = Block::rand_spending_transaction(&ctx);
+        let output_id = if let Input::Utxo(output_id) = input {
+            output_id
+        } else {
+            unreachable!();
+        };
+        let parents = block.parents.clone();
+        let raw = bee_block_stardust::rand::bytes::rand_bytes(100);
+        let metadata = BlockMetadata {
+            parents,
+            is_solid: true,
+            should_promote: false,
+            should_reattach: false,
+            referenced_by_milestone_index: 1.into(),
+            milestone_index: 0.into(),
+            inclusion_state: LedgerInclusionState::Included,
+            conflict_reason: ConflictReason::None,
+            white_flag_index: 0u32,
+        };
+
+        let blocks = vec![(block_id, block.clone(), raw, metadata)];
+        collection.insert_blocks_with_metadata(blocks).await.unwrap();
+
+        assert_eq!(
+            collection
+                .get_spending_transaction(&output_id)
+                .await
+                .unwrap()
+                .map(|b| b.protocol_version),
+            Some(block.protocol_version)
+        );
+        assert_eq!(
+            collection
+                .get_spending_transaction(&output_id)
+                .await
+                .unwrap()
+                .map(|b| b.parents),
+            Some(block.parents)
+        );
+        // assert_eq!(collection.get_spending_transaction(&output_id).await.unwrap().map(|b| b.payload),
+        // Some(block.payload));
```

Comment:
What's up with this?

Reply:
Can we open an issue for it, so that it is not blocking this PR? Maybe add a TODO with the link to the issue to the file as well.

Reply:
it's fixed now by limiting the spending transaction to only have one output, but I can open an issue so we can fix this later. The test is valid though as it proves what it's supposed to even if there's only 1 output.